tests/descriptor_tests/test_api.py

```diff
@@ -144,6 +145,7 @@ def test_alt_cache_root(self):
         bundle_root = os.path.join(tempfile.gettempdir(), uuid.uuid4().hex)
 
         os.makedirs(bundle_root)
+        self.addCleanup(shutil.rmtree, bundle_root)
```

Comment:
In most other places in the code, we would just implement a tearDown and do the deletion there. In order to keep things consistent, I would prefer that approach rather than using `addCleanup` which we aren't using anywhere else. Could we just make `bundle_root` -> `self. bundle_root` and then add a teardown?

Reply:
Actually, we're extensively using addCleanup in the TankTestBase now ;)

Reply:
sorry, looks like my grep-fu failed me somehow. It's used in a couple of places already, often in conjunction with the patcher. It also makes perfect sense to use in the zip file case later on, but wondering if this particular case might be clearer in the teardown? I'm happy with whatever you guys think is right.